drivers/flash/flash_ite_it8xxx2.c

```diff
@@ -0,0 +1,582 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_flash_controller
+#define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
+
+#define FLASH_WRITE_BLK_SZ DT_PROP(SOC_NV_FLASH_NODE, write_block_size)
+#define FLASH_ERASE_BLK_SZ DT_PROP(SOC_NV_FLASH_NODE, erase_block_size)
+
+#include <device.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <kernel.h>
+#include <linker/linker-defs.h>
+#include <soc.h>
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_ite_it8xxx2);
+
+extern char _ram_code_start;
+#define FLASH_RAMCODE_START ((uint32_t)&_ram_code_start)
+#define __ram_code __attribute__((section(".__ram_code")))
+
+#define FLASH_IT8XXX2_REG_BASE \
+		((struct flash_it8xxx2_regs *)(DT_INST_REG_ADDR(0)))
+
+#define DEV_DATA(dev) \
+	((struct flash_it8xxx2_dev_data *const)(dev)->data)
+
+struct flash_it8xxx2_dev_data {
+	struct k_sem sem;
+	int all_protected;
+	int flash_static_cache_enabled;
+};
+
+/*
+ * One page program instruction allows maximum 256 bytes (a page) of data
+ * to be programmed.
+ */
+#define CHIP_FLASH_WRITE_PAGE_MAX_SIZE 256
+/* Program is run directly from storage */
+#define CHIP_MAPPED_STORAGE_BASE       DT_REG_ADDR(DT_NODELABEL(flash0))
+/* flash size */
+#define CHIP_FLASH_SIZE_BYTES          DT_REG_SIZE(DT_NODELABEL(flash0))
+/* protect bank size */
+#define CHIP_FLASH_BANK_SIZE           0x00001000
+/* base+0000h~base+0FFF */
+#define CHIP_RAMCODE_BASE              0x80100000
+
+/*
+ * This is the block size of the ILM on the it8xxx2 chip.
+ * The ILM for static code cache, CPU fetch instruction from
+ * ILM(ILM -> CPU)instead of flash(flash ->  I-Cache -> CPU) if enabled.
+ */
+#define IT8XXX2_ILM_BLOCK_SIZE 0x00001000
+
+/* page program command  */
+#define FLASH_CMD_PAGE_WRITE   0x2
+/* ector erase command (erase size is 4KB) */
+#define FLASH_CMD_SECTOR_ERASE 0x20
+/* command for flash write */
+#define FLASH_CMD_WRITE        FLASH_CMD_PAGE_WRITE
+/* Write status register */
+#define FLASH_CMD_WRSR         0x01
+/* Write disable */
+#define FLASH_CMD_WRDI         0x04
+/* Write enable */
+#define FLASH_CMD_WREN         0x06
+/* Read status register */
+#define FLASH_CMD_RS           0x05
+
+#define FWP_REG(bank) (bank / 8)
+#define FWP_MASK(bank) (1 << (bank % 8))
+
+enum flash_wp_interface {
+	FLASH_WP_HOST = 0x01,
+	FLASH_WP_DBGR = 0x02,
+	FLASH_WP_EC = 0x04,
+};
+
+enum flash_status_mask {
+	FLASH_SR_NO_BUSY = 0,
+	/* Internal write operation is in progress */
+	FLASH_SR_BUSY = 0x01,
+	/* Device is memory Write enabled */
+	FLASH_SR_WEL = 0x02,
+
+	FLASH_SR_ALL = (FLASH_SR_BUSY | FLASH_SR_WEL),
+};
+
+static const struct flash_parameters flash_it8xxx2_parameters = {
+	.write_block_size = FLASH_WRITE_BLK_SZ,
+	.erase_value = 0xff,
+};
+
+void __ram_code ramcode_reset_i_cache(void)
+{
+	/* I-Cache tag sram reset */
+	IT83XX_GCTRL_MCCR |= 0x10;
+
+	IT83XX_GCTRL_MCCR &= ~0x10;
+}
```

Comment:
Can you please scan through all of the magic numbers through out this file and see if it would be more readable for these to meaningful defines?

Reply:
Done